system/security/securesocket/securesocket.cpp

```diff
@@ -362,13 +365,42 @@ class CSecureSocket : implements ISecureSocket, public CInterface
                 errbuf[511] = '\0';
                 DBGLOG("SSL_peek (avail_read) returns error %d - %s", ret, errbuf);
             }
+            -------------------------------- */
         }
         return 0;
     }
 
-    virtual size32_t write_multiple(unsigned num,const void **buf, size32_t *size)
+    virtual size32_t write_multiple(unsigned num, const void **buf, size32_t *size)
     {
-        throw MakeStringException(-1, "CSecureSocket::write_multiple: not implemented");
+        if (num == 0)
```

Comment:
The code is correct, but I'm not convinced either of the special cases are worth it. Probably outweighed by complication/loss of instruction cache.